### ISSUE TITLE
Report false negative structures here

### ISSUE THREAD

**Issue by BigAlanM** [toolbox4minecraft/amidst#117]
What that means is:
Surface structures that DON'T show up on the Amidst map that someone has reported and has been verified as existing in a vanilla version of Minecraft map.

To start off I just read a post on the Seeds forum that the map for the seed "546868080" does not show a savanna village at 36,676. 

I created the world in PCMCv1.8.9 and checked that the village does, in fact, exist.
All of the well is in the savanna biome but there is a river biome starting 1 meter to the east of it.
All of the village is west of the well.

I do, on occasion, find ~10% false positives (almost always villages).
That, I know, we have to live with.

(As an aside TOTALLY unrelated to false negatives, I've just noticed that Amidst 4.0 uses a different location to define surface structure coordinates than it did before.
The reported coordinates are closer to the center of the defining object than they used to be.)

**Comment by stefandollase** [comment]
I added more formatting to your opening post, so this has a chance to become overseeable when there are multiple reports.

---

Thankfully, @Treer increased the accuracy of structures in AmidstExporter. These changes were ported back to Amidst.

Do I understand you correctly, that we should move the icon of strongholds 2 blocks further in positive x and z direction?

**Comment by BigAlanM** [comment]
Well, moving the displayed value of Strongholds 2 east and 2 south WOULD cause someone digging straight down at that X,Z location to actually hit the NW corner stone blocks of the stairwell and if you continue digging down you will enter the stairwell without falling any distance.
Currently it's quite possible to actually get all the way to bedrock without hitting any evidence of the stronghold complex at all.
BTW: The location shown for Abandoned Mines is a point inside a large, fairly high, dirt floored room with tunnels usually leading off in all four directions. Digging straight down at this location without flying will definitely result in a fall with some damage. I haven't checked enough mine locations to establish if the coordinates indicate a specific point in this room. I assume they do, I just don't know what it is.
I've been telling people for quite some time to dig, at least that far away from the indicated location for a couple of years.
I've always assumed that Amidst was reporting the X & Z coordinates produced by the world generator code without any modification.  If that's the case then in 1.9 Mojang has changed how, at least, Desert Temples are located.  In prior versions the DT location was defined by the NW corner block of the temple.  What I've found on the few I've checked it's a point closer to the center hole in the roof but still NW of it.
I don't know if the created structures rotate around this point or not. IE: if a corner defines a structure's location does the direction it's facing change which corner this point is on relative to the structure.  My experience is that it doesn't.

**Comment by BlueSteelAUS** [comment]
shouldn't the stronghold point to enderportal ? is that even possible ?

**Comment by BigAlanM** [comment]
No, IFAIK, the locations of the end portal rooms like those for spawn dungeons aren't defined until the actual chunk data is finalized and written to disk during game play.

**Comment by Treer** [comment]
> IFAIK, the locations of the end portal rooms like those for spawn dungeons aren't defined until the actual chunk data is finalized and written to disk during game play.

This is my understanding as well.

With regard to Mineshafts, the dirt room is the "genesis structure" of the mineshaft from which the shafts grow out from. If you find more than one dirt room in a mineshaft it means two mineshafts have grown together (the equivalent structure in a village is the well). I tried to get AE as close to the center of the dirt room as possible because it's the center and there might not be a shaft in every direction from it.

With regard to the Strongholds. In AE I wanted to adjust the coords so that digging straight down would always hit the stronghold, but using trial and error I noticed that the stronghold isn't always in the same direction from the coords. The End cities are similiar in this regard - the coords are normally at one corner, but it's not always the same corner. So try 5 to 8 different strongholds, I think you'll find it sometimes isn't south-east to the coords. This is based on trial and error though, I haven't tried to figure out how it's generated.

**Comment by BigAlanM** [comment]
A thread was just started in the PC Seeds forum that presented a world created with Snapshot 16w21b that has a Taiga village which, of course, Amidst doesn't predict.
I tried Amidst v4.2b with my Snapshot 16w21b profile and the village isn't shown, which is as I expected.
Here's the thread link: http://www.minecraftforum.net/forums/minecraft-discussion/seeds/2682970-taiga-village-seed-1725599750689066986
Apparently the next PCMC release will add Taiga biomes to the list of biomes where villages can exist and also allow villages to be created that expand into other biomes. That is be created close to edge of a allowable biome.

**Comment by stefandollase** [comment]
@BigAlanM Thanks, you are right. We did not yet implement villages in other biomes, like they are generated in Minecraft 1.10. I created #177 for it.

**Comment by moulins** [comment]
Issue #423 reported another false negative; I've added it to the first post.

**Comment by fwiffo** [comment]
Could this just be caused by the lower resolution biome data (which also causes false positives)?

**Comment by fwiffo** [comment]
-680543071556675343 in 18w32a. False negative desert village at 1893 -347. This one needs to be seen to be believed.
![2018-08-12_17 47 55](https://user-images.githubusercontent.com/169511/44008400-147aeea6-9e58-11e8-90f7-9d0cb2d1c367.png)

**Comment by moulins** [comment]
Huh, this is a weird village! It seems the well generated at the border with the River biome; and this only happens in 1.13, not in 1.12 or older.

By disabling the [`VillageAlgorithm`](https://github.com/toolbox4minecraft/amidst/blob/master/src/main/java/amidst/mojangapi/world/icon/locationchecker/VillageAlgorithm.java) check, I can get the village (and the one in seed `-3086593819764553073`) to show up on Amidst. However, I don't know if it's something we want to do; it may add false positives, and I don't know which Minecraft versions should be affected.

Thanks for the report!

**Comment by fwiffo** [comment]
Ice spikes is an M type biome, so its outline changed in 1.13. In 1.12 and older the ice spikes go across the river a bit in this area, but in 1.13 it conforms to the river exactly. It also doesn't leak into the mesa areas creating mesa Bryce any more.

**Comment by fwiffo** [comment]
Found a false-negative ocean monument in 1.13.1; seed 6546194431633768665 at roughly -2368 -288. It clips into the neighboring terrain a fair bit.
![2018-09-11_01 42 02](https://user-images.githubusercontent.com/169511/45348741-4b54b600-b564-11e8-9ccb-983d11fd6739.png)

This one also fools Cubitect's structure/biome detection code (which is a 100% completely different implementation), which was not the case with the village above.

**Comment by moulins** [comment]
Thanks for the report! It seems this Monument first appears in version 18w09a.

**Comment by yaboiarb** [comment]
Help, if this is unrelated I'll delete but amidst shows me a pillager outpost that when I go to the exact location of it just isn't there. It's pretty much showing me a fake pillager outpost.
